Hypermedia.AspNetCore.Siren/HypermediaResourceFilter.cs

```diff
@@ -1,20 +1,27 @@
 ﻿namespace Hypermedia.AspNetCore.Siren
 {
-    using System.Linq;
-    using Entities.Builder;
+    using Hypermedia.AspNetCore.Siren.Resources;
     using Microsoft.AspNetCore.Mvc;
     using Microsoft.AspNetCore.Mvc.Filters;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Linq;
+    using System.Threading.Tasks;
 
-    internal class HypermediaResourceFilter : IResultFilter
+    internal class HypermediaResourceFilter : IAsyncResultFilter
     {
-        private readonly EntityBuilder _builder;
+        private readonly IServiceScopeFactory _factory;
 
-        public HypermediaResourceFilter(EntityBuilder builder)
+        public HypermediaResourceFilter(IServiceScopeFactory factory)
         {
-            this._builder = builder;
+            this._factory = factory;
         }
 
-        public void OnResultExecuting(ResultExecutingContext context)
+        public Task OnResultExecutingAsync(ResultExecutingContext context)
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next)
         {
             if (!(context.Result is ObjectResult objectResult))
```

Comment:
We could unite these conditional statements as they result in the same behavior.

Reply:
I cannot do this, because
``` csharp
if (!(context.Result is ObjectResult objectResult) &&
    !(objectResult.Value is IHypermediaResource resource)) 
// Unassigned local variable objectResult 
{
       return;
}
```
Further more, it is preferred to separate the clauses of any if when returning out of the method. This makes it easier to see where your actual exit points from the method are.